gdal/port/cpl_vsil_curl.cpp

```diff
@@ -2300,17 +2282,48 @@ int       VSICurlHandle::Close()
 /************************************************************************/
 
 VSICurlFilesystemHandler::VSICurlFilesystemHandler():
-    oRegionCache{static_cast<size_t>(N_MAX_REGIONS)}
+    oRegionCache{static_cast<size_t>(N_MAX_REGIONS)},
+    oCacheFileProp{1024},
```

Comment:
I'd be tempted to make this bigger, especially if we're caching files _not_ existing (eg. `foo.tif.aux.xml` or the other extensions GDAL searches for). Feels like a directory with a couple of hundred images in it would exhaust the cache?

Looks like (depending on compiler class-member handling and assuming `osRedirectURL` is null mostly) a `FileProp` should be <50 bytes on a 64bit system, so 1K entries is <64KB of memory...